### PR TITLE
Gtfs realtime

### PR DESCRIPTION
**Pull request by tylerprete**

### CODE REVIEW

**Comment by tylerprete** [comment]
Probably not quite ready to merge -- but ready for review. @ryle-goehausen especially can you review DataPollingManager? Am I submitting messages to a channel correctly?

File: src/main/scala/org/constellation/p2p/Download.scala [hunk 1]
```diff
@@ -89,7 +88,24 @@ class SnapshotsProcessor(downloadSnapshot: (String, Iterable[APIClient]) => IO[S
       .flatMap(acceptSnapshot)
   }
 
+  private def downloadSnapshot(hash: String, pool: Array[EnhancedAPIClient]): IO[StoredSnapshot] = {
```
Comment (marcinwadon) [inline src/main/scala/org/constellation/p2p/Download.scala]: Why it's needed here if `downloadSnapshot` is passed as an argument and should be abstracted from here?

**Comment by Nikolaj-K** [comment]
I had no better idea, but I was wondering if there's something more specific than "Enhanced". Something from which the differentiation to the APIClient explains itself.

**Comment by tylerprete** [comment]
I agree -- totally open to names. Would've liked to use something from the
company but the name is so long. ConstellationAPIClient is quite the
mouthful but I guess I could use that.

The idea here is to distinguish between an APIClient with no functionality
specific to our app, as I plan to pull it out to it's own package
eventually (along with this new transit stuff).

Then the Enhanced (or whatever) can have extra stuff like an id, or
constellation specific methods.

**Comment by Nikolaj-K** [comment]
Product management at my work had us remove the company name from the various interfaces and individual tools in favor of bold but specific names (e.g. `VisionCore` and [`SlamCore`](https://en.wikipedia.org/wiki/Simultaneous_localization_and_mapping)) and branding of apps also goes in that direction (`Scan`, `Guide`, etc.). Core is generally a good word to signify a hierarchy.

Here's some thoughts, that depend on how they will be used as API as well as further code in this repo as well as externals:

* Regarding the `APIClient` : What you could do is tag this one instead. In case it will always be wrapped, you can make that one the `APIClientBase` (a naming akin to the Factory patters). But if said `APIClient` will further down the road be used directly as well, you can make it the `BaseAPIClient`. 

* Regarding the `EnhancedAPIClient`: If that one will always be used as a starting point for other clients, you make it the `APICoreClient` or (if the base one is renamed) name the "enhanced one"  as `APIClient`. If the enhanced one will actually instead most likely not further be wrapped but used often directy, you can also make it the `CoreAPIClient`. 

There's also something to be said about the addon of "Client" in the first place, i.e. `APIBase` and `CoreAPI` (and making room for `ConstellationAPI` or ). Same goes for, instead, going with something along the lines of `xxxInterface`, in case there's more functionality in it than application programming.

File: src/main/scala/org/constellation/p2p/Download.scala (hunk 1 above)
Comment (tylerprete) [reply src/main/scala/org/constellation/p2p/Download.scala]: I don't quite know what you mean Marcin. This is really the same as before, I just renamed APIClient.

**Comment by tylerprete** [comment]
Good feedback @Nikolaj-K. Perhaps their is a better name than Client, but for now I'll go with your idea of using Base for the bottom layer and APIClient for top.

File: src/main/scala/org/constellation/p2p/Download.scala (hunk 1 above)
Comment (marcinwadon) [reply src/main/scala/org/constellation/p2p/Download.scala]: Not really. In the changes I can see that you copied part of the code and pasted here. You can see it on the diff. And I think this is just some dummy mistake :) but there shouldn’t be this code

Comment (tylerprete) [reply src/main/scala/org/constellation/p2p/Download.scala]: Interesting. Maybe a merge issue? I don't remember copying this on purpose. But regardless, I'll get rid of it if it's not needed. Good catch.

**Pull request merged**